recipes/rdma-core/all/conanfile.py

```diff
@@ -0,0 +1,133 @@
+import os
+import re
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout, CMakeDeps
+from conan.tools.env import VirtualBuildEnv
+from conan.tools.files import get, copy, rmdir, load, save, replace_in_file
+from conan.tools.gnu import PkgConfigDeps
+
+required_conan_version = ">=1.53.0"
+
+class PackageConan(ConanFile):
+    name = "rdma-core"
+    description = ("RDMA core userspace libraries and daemons. "
+                   "Provides userspace components for the Linux Kernel's drivers/infiniband subsystem.")
+    license = ("GPL-2.0", "Linux-OpenIB", "BSD-2-Clause")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/linux-rdma/rdma-core"
+    topics = ("linux-kernel", "rdma", "infiniband", "iwarp", "roce", "kernel-rdma-drivers",
+              "libefa", "libibmad", "libibnetdisc", "libibumad", "libibverbs", "libmana",
+              "libmlx4", "libmlx5", "librdmacm")
+
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "build_libefa": [True, False],
+        "build_libibnetdisc": [True, False],
+        "build_libmana": [True, False],
+        "build_libmlx4": [True, False],
+        "build_libmlx5": [True, False],
+        "build_librdmacm": [True, False],
+    }
+    default_options = {
+        "build_libefa": True,
+        "build_libibnetdisc": True,
+        "build_libmana": True,
+        "build_libmlx4": True,
+        "build_libmlx5": False,
+        "build_librdmacm": False,
+    }
+
+    def configure(self):
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libnl/3.8.0")
+        if self.settings.os in ["Linux", "FreeBSD"]:
+            self.requires("libudev/system")
```

Comment:
The validation is already in the `validate()` function.

```suggestion
        self.requires("libudev/system")
```